### PR TITLE
refactor: update jobs pages to new design

### LINKED ISSUE #113: Update jobs page layout
## Description

The figma design for this is available [here](https://www.figma.com/file/msxmKajRPLYiDDmsP60Ndh/Design-System-v1.1?type=design&node-id=2279-105883&mode=design&t=LPbiXc0jHu307v21-0).

## Tasks

- [x] Introduce index page under `/jobs`
- [x] Decide on how to treat the apply shared part (maybe separate markdown file to load it from) and remove it from each md file
- [x] Adjust the layout

### PR DESCRIPTION
**Pull request by lukicenturi**
Closes #113

### CODE REVIEW

**Comment by lukicenturi** [comment]
To test using the new markdown, in `nuxt.config.ts` line 165 change to 
```
        repo: 'lukicenturi/rotki.com',
        branch: 'jobs',
```

**Review by tewshi — CHANGES_REQUESTED** [review]

File: pages/jobs/JobTag.vue
[the remark this answers is not in the document]
Comment (lukicenturi) [reply pages/jobs/JobTag.vue:1]: you mean the whole component? It use <RuiChip/ > inside.

[the remark this answers is not in the document]
Comment (tewshi) [reply pages/jobs/JobTag.vue:1]: I meant to move this file to components

[the remark this answers is not in the document]
Comment (lukicenturi) [reply pages/jobs/JobTag.vue:1]: oh sorry, see that now, updated

**Review by tewshi — APPROVED** [review]
lgtm!

**Review by kelsos — APPROVED** [review]

**Pull request merged**